### PR TITLE
Fix missing push nds for headless pure http svc

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
This is a regression  from #48235, cc @GregHanson

**Comment by hzxuzhonghu** [comment]
@ramaraochavali

File: pilot/pkg/serviceregistry/serviceregistry_test.go [hunk 1]
```diff
@@ -241,6 +241,11 @@ func TestWorkloadInstances(t *testing.T) {
 		makePod(t, kube, pod)
 		createEndpoints(t, kube, service.Name, namespace, []v1.EndpointPort{{Name: "http", Port: 80}}, []string{pod.Status.PodIP})
 		fx.WaitOrFail(t, "eds")
+		// headless service update must trigger nds push.
+		ev := fx.WaitOrFail(t, "xds")
+		if !ev.Reason.Has(model.HeadlessEndpointUpdate) {
+			t.Fatalf("xds push reason doesnot contain %v", model.HeadlessEndpointUpdate)
```
Comment (ymesika) [inline pilot/pkg/serviceregistry/serviceregistry_test.go:247]: Very nit: `doesn't` / `does not`

File: pilot/pkg/serviceregistry/serviceregistry_test.go [hunk 2]
```diff
@@ -257,7 +262,11 @@ func TestWorkloadInstances(t *testing.T) {
 		makePod(t, kube, pod)
 		createEndpoints(t, kube, service.Name, namespace, []v1.EndpointPort{{Name: "tcp", Port: 70}}, []string{pod.Status.PodIP})
 		fx.WaitOrFail(t, "eds")
-		fx.WaitOrFail(t, "xds full")
+		ev := fx.WaitOrFail(t, "xds full")
+		// headless service update must trigger nds push.
+		if !ev.Reason.Has(model.HeadlessEndpointUpdate) {
+			t.Fatalf("xds push reason doesnot contain %v", model.HeadlessEndpointUpdate)
```
Comment (ymesika) [inline pilot/pkg/serviceregistry/serviceregistry_test.go:268]: ditto

File: pilot/pkg/serviceregistry/serviceregistry_test.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/serviceregistry_test.go:247]: ack, will fix

**Review by ramaraochavali — APPROVED** [review]

**Pull request merged**